lib/features.js

```diff
@@ -0,0 +1,105 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+'use strict'
+
+const hash = require('sha1')
```

Comment:
Thanks both. I'm guilty of laziness here, I used abatar as the reference implementation and didn't didn't think through many of the details. Since these functions are called pretty frequently (albeit not in inside any loops), it's definitely worth taking the time to benchmark the different options.